### PR TITLE
Fix remaining minor input/output issues with TG-Llama3 vLLM integration

### PR DESCRIPTION
**Pull request by skhorasganiTT**
### Ticket
N/A

### Problem description
- There were some minor input and output processing issues when running `LlamaGenerator` with TG

### What's changed
- Fixed token input processing and logit output processing in functions used by `LlamaGenerator` (currently used by vLLM but not model demo)
- Sliced output logits in demo to vocab size (for TG was returning padded vocab size but did not affect output tokens)

### Checklist
- [x] Post commit CI passes
- [x] Blackhole Post commit (if applicable)
- [x] Model regression CI testing passes (if applicable)
- [x] Device performance regression CI testing passes (if applicable)
- [x] **(For models and ops writers)** Full [new models](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml) tests passes
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by skhorasganiTT** [comment]
T3K demo & frequent tests: https://github.com/tenstorrent/tt-metal/actions/runs/12602265797

File: models/demos/llama3/README.md [hunk 1]
```diff
@@ -25,9 +25,7 @@ The main reason for a long context length not fitting on device is lack of memor
 | Llama3.2-3B  | 32k tokens    | 128k tokens   | 128k tokens    | 128k tokens |
 | Llama3.1-8B  | 16k tokens    | 64k tokens    | 128k tokens    | 128k tokens |
 | Llama3.2-11B | 16k tokens    | 64k tokens    | 128k tokens    | 128k tokens |
-| Llama3.1-70B | Not supported | Not supported | 64k tokens [1] | 128k tokens |
-
-[1] Although longer prefill context-lengths are not supported due to model size and available memory, you can still decode (generate) tokens up to a maximum of 128k tokens.
+| Llama3.1-70B | Not supported | Not supported | 128k tokens    | 128k tokens |
```
Comment (cglagovichTT) [inline models/demos/llama3/README.md:28]: is this tested?

**Review by cglagovichTT — APPROVED** [review]

File: models/demos/llama3/README.md (hunk 1 above)
Comment (skhorasganiTT) [reply models/demos/llama3/README.md]: I have removed the update to the readme, it seems like we still have an issue with some long sequence lengths. We can address that separately.
